data_linter/validation.py

```diff
@@ -510,17 +511,24 @@ def run_validation(config: Union[str, dict] = "config.yaml"):
         log.info("Running validation")
         validate_data(config)
     except Exception as e:
-        upload_log(body=log_stringio.getvalue(), s3_path=log_path)
-        log_msg = (
-            f"Unexpected error hit. Uploading log to {log_path}. Before raising error."
-        )
-        error_msg = str(e)
+        if log_path:
+            log_msg = (
+                f"Unexpected error. Uploading log to {log_path} before raising error."
+            )
+            error_msg = str(e)
 
-        log.error(log_msg)
-        log.error(error_msg)
+            log.error(log_msg)
+            log.error(error_msg)
 
-        upload_log(body=log_stringio.getvalue(), s3_path=log_path)
+            upload_log(body=log_stringio.getvalue(), s3_path=log_path)
+        else:
+            log.error("An error occurred but no log path registered, "
+                      "so cannot upload to log S3.")
 
         raise e.with_traceback(e.__traceback__)
     else:
-        upload_log(body=log_stringio.getvalue(), s3_path=log_path)
+        if log_path:
+            upload_log(body=log_stringio.getvalue(), s3_path=log_path)
+        else:
+            log.error("Something went wrong but no log path was given, "
+                      "so log won't be uploaded")
```

Comment:
Might be worth wrapping this into a function. I got confused by the control logic here. Might be cleaner if this function just did something like:

```python
def try_upload_log_to_s3(log, log_string_io, log_path):
    # Then put the if logic in here (if log_path) log.info writing to S3 then upload. Otherwise log error as path is none and then do nothing. 
```

That way this can then go in both parts of the except and else

Reply:
ok fine